src/charts/donut/donut-pct-chart-directive.js

```diff
@@ -0,0 +1,354 @@
+       'tooltipFn': function (d) {
+         return '<span class="donut-tooltip-pf"style="white-space: nowrap;">' +
+                   d[0].value + ' ' + d[0].name +
+                '</span>';
+       },
+       'centerLabelFn': function (scope) {
+         return '<tspan dy="0" x="0" class="donut-title-big-pf">' +
+                  scope.data.available + '</tspan>' +
+                '<tspan dy="20" x="0" class="donut-title-small-pf">' +
+                  'Free' +
+                '</tspan>';
+       }
+     };
+
+     $scope.custData = {
+         'used': '670',
+         'total': '1000'
+     };
+
+   };
+ </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfDonutPctChart', ['ChartsMixin', '$timeout',
+  function(chartsMixin, $timeout) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        id: '@',
+        config: '=',
+        data: '=',
+        centerLabel: '@'
+      },
+      replace: true,
+      templateUrl: 'charts/donut/donut-pct-chart.html',
+      controller: ['$scope',
+        function($scope) {
+
+          $scope.donutChartId = 'donutChart';
+          if ($scope.id) {
+            $scope.donutChartId = $scope.id + $scope.donutChartId;
+          }
+
+          $scope.updateAvailable = function(){
+            $scope.data.available = $scope.data.total - $scope.data.used;
+          };
+
+          if ($scope.data.available === undefined) {
+            $scope.updateAvailable();
+          }
+
+          $scope.getStatusColor = function(used, thresholds) {
+            if (!thresholds){
+              return '#0088CE';
+            } else {
+              if (used >= thresholds.error) {
+                return '#CC0000';
+              } else if (used >= thresholds.warning) {
+                return '#EC7A08';
+              } else {
+                return '#0088CE';
+              }
+            }
+          };
+
+          $scope.statusDonutColor = function(scope) {
+            var color = {
+              pattern: []
+            };
+            var percentUsed = scope.data.used / scope.data.total * 100.0;
+            color.pattern[0] = $scope.getStatusColor(percentUsed, scope.config.thresholds);
+            color.pattern[1] = '#D1D1D1';
+            return color;
+          };
+
+          var donutTooltip = function(scope) {
+            return {
+              contents: function (d) {
+                if (scope.config.tooltipFn) {
+                  var tootipHtml = '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                                        scope.config.tooltipFn(d) +
+                                   '</span>';
+                  return tootipHtml;
+                } else {
+                  return '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                            Math.round(d[0].ratio * 100) + '%' + ' ' + $scope.config.units + ' ' + d[0].name +
+                         '</span>';
+                }
```

Comment:
> Can we allow these discrepancies for this PR and fix them if/when #36 gets accepted? If we decide to enforce these rules, then these issues can be fixed at that time.

Definitely.  I don't want to hold up this PR on something as of yet undecided.

I'll start a thread on the mailing list about this because it's outside of the context of this particular PR and when this is merged I have no problem running my regexes again to fix the spacing for #36.

> ... comment out the outer function, then 'grunt build'...

What if we used the grunt live reload server like most other grunt-based projects?  Is there a reason why we are building every single time we need to test a change?

P.S.: Check out [my website's gruntfile](https://github.com/waldenraines/waldenrainesdotcom/blob/master/Gruntfile.js#L28-L61) for an example (based off of yeoman's [generator-angular](https://github.com/yeoman/generator-angular)), particularly the watch config.  It runs  jshint on javascript files when they are changed but that doesn't kill the webserver.  It also reloads the web browser on changes and runs tests on test file changes.